src/browser/base/content/ZenUIManager.mjs

```diff
@@ -583,4 +585,95 @@ var gZenVerticalTabsManager = {
     }
     target.appendChild(child);
   },
+
+  //_insertDoubleClickListenerPinnedTabs() {
+  //  const tabs = gBrowser.tabs;
+  //  for (const tab of tabs) {
+  //    tab.addEventListener('dblclick', this.renameTabStart.bind(this));
+  //  }
+  //},
+
+  renameTabKeydown(event) {
+    if (event.key === 'Enter') {
+      let label = this._tabEdited.querySelector('.tab-label-container-editing');
+      let input = this._tabEdited.querySelector('#tab-label-input');
+      let newName = input.value.trim();
+
+      // Check if name is blank, reset if so
+      if (newName) {
+        this._tabEdited.label = newName;
+        this._tabEdited.setAttribute('zen-has-static-label', 'true');
+      } else {
+        // If the page is loaded, get the title of the page. Otherwise, keep name as is
+        this._tabEdited.label = gBrowser.getBrowserForTab(this._tabEdited).contentTitle || this._tabEdited.label;
+        // If the page had a title, reset the zen-has-static-label attribute
+        if (gBrowser.getBrowserForTab(this._tabEdited).contentTitle) {
+          this._tabEdited.removeAttribute('zen-has-static-label');
+        }
+      }
+
+      this._tabEdited.querySelector('.tab-editor-container').remove();
+      label.style.display = '';
+      label.className = label.className.replace(' tab-label-container-editing', '');
+      document.removeEventListener('click', this.renameTabHalt.bind(this));
+
+      this._tabEdited = null;
+    } else if (event.key === 'Escape') {
+      let label = this._tabEdited.querySelector('.tab-label-container-editing');
+      this._tabEdited.querySelector('.tab-editor-container').remove();
+
+      label.style.display = '';
+      label.className = label.className.replace(' tab-label-container-editing', '');
+      document.removeEventListener('click', this.renameTabHalt.bind(this));
```

Comment:
`removeEventListener` doesn't actually remove anything if you are using `bind`... Id do on the init function `this._renameTabHalt = this.renameTabHalt.bind(this)` and then `removeEventListener(..., this._renameTabHalt)` and same for `addEventListener`